pkg/helmreconciler/rendering.go

```diff
@@ -146,41 +146,32 @@ func unmarshalAndValidateICPS(icpsYAML string) (*v1alpha2.IstioControlPlaneSpec,
 	return icps, nil
 }
 
-func (h *HelmReconciler) ProcessManifests(manifests []manifest.Manifest) error {
-	allErrors := []error{}
-	origLogger := h.logger
-	defer func() { h.logger = origLogger }()
-	for _, manifest := range manifests {
-		h.logger = origLogger.WithValues("manifest", manifest.Name)
-		if !strings.HasSuffix(manifest.Name, ".yaml") {
-			h.logger.V(2).Info("Skipping rendering of manifest")
+func (h *HelmReconciler) ProcessManifest(manifest manifest.Manifest) error {
+	var errs []error
+	log.Info("Processing resources from manifest")
+	// split the manifest into individual objects
+	objects := releaseutil.SplitManifests(manifest.Content)
+	for _, raw := range objects {
+		rawJSON, err := yaml.YAMLToJSON([]byte(raw))
+		if err != nil {
+			h.logger.Error(err, "unable to convert raw data to JSON")
```

Comment:
I'm unclear if a log message is necessary here.  Better would be to produce an error string and append that to errs in line 158.

Reply:
Done.